### PR TITLE
 Generate deb package on release #94 

### LINKED ISSUE #94: Generate `deb` package on release
Generate `deb` packages on release using [`cargo-deb`](https://github.com/kornelski/cargo-deb) or similar

### PR DESCRIPTION
**Pull request by Marwanmhks**
Closes #94

### CODE REVIEW

**Comment by fujiapple852** [comment]
@Marwanmhks are you able to try this out on your fork?  It’s tricky to iterate on it in this repo as it’ll only work from master and will create spurious releases as we test. 

Thanks again for helping with this.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -97,6 +97,27 @@ jobs:
           tar czf "$staging.tar.gz" "$staging"
           echo "ASSET=$staging.tar.gz" >> $GITHUB_ENV
 
+      - name: Create Debian package
+        shell: bash
+        if: startsWith(matrix.job.os, 'ubuntu')
```
Comment (fujiapple852) [inline .github/workflows/release.yml:102]: It looks like cargo-deb supports passing a —target flag to create deb packages for each platform, would it be possible to do that?  Not critical for the first version I’d say, nice to have, we can always extend it later.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -97,6 +97,27 @@ jobs:
...
+            x86_64-*-linux-*) DPKG_ARCH=amd64 ;;
+            *) DPKG_ARCH=notset ;;
+          esac;
+          PROJECT_NAME = "trippy"
+          PKG_BASENAME = ${PROJECT_NAME}-v${TRIP_VERSION}-${{ matrix.job.target }}
+          DPKG_NAME="${PROJECT_NAME}_${TRIP_VERSION}_${DPKG_ARCH}.deb"
+          DPKG_PATH="debian/${PKG_BASENAME}.deb"
+          DPKG_PATH="debian/${DPKG_NAME}"
+          echo ::set-output name=DPKG_NAME::${DPKG_NAME}
+          echo ::set-output name=DPKG_PATH::${DPKG_PATH}
```
Comment (fujiapple852) [inline .github/workflows/release.yml:119]: I think there needs to be another step to upload the generated deb artefact(s)?

**Comment by Marwanmhks** [comment]
okay let me check it out and get back to you

**Comment by Marwanmhks** [comment]
Okay so I was able to automate the deb packaging upon release but I am have a problem with the artifact upload, I would appreciate any hints.

**Comment by fujiapple852** [comment]
@Marwanmhks I think you almost had it, nice work.  

I had a try on a fork (see [here](https://github.com/zoie-bg/trippy/releases/tag/1.0.7)) and was able to get it working with the following:

```
      - name: Build Debian package
        if: startsWith(matrix.build, 'linux-gnu')
        run: |
          cargo install cargo-deb
          cargo deb --target ${{ matrix.target }} --deb-version ${{ needs.create-release.outputs.trip_version }}
          case ${{ matrix.target }} in
            aarch64-*-linux-*) DPKG_ARCH=arm64 ;;
            arm-*-linux-*hf) DPKG_ARCH=armhf ;;
            i686-*-linux-*) DPKG_ARCH=i686 ;;
            x86_64-*-linux-*) DPKG_ARCH=amd64 ;;
            *) DPKG_ARCH=notset ;;
          esac;
          echo "DPKG_ARCH=${DPKG_ARCH}" >> $GITHUB_ENV
          
      - name: Upload assets to Github
        if: startsWith(matrix.build, 'linux-gnu')
        uses: actions/upload-release-asset@v1.0.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ needs.create-release.outputs.upload_url }}
          asset_content_type: application/vnd.debian.binary-package
          asset_path: target/${{ matrix.target }}/debian/trippy_${{ needs.create-release.outputs.trip_version }}_${{ env.DPKG_ARCH }}.deb
          asset_name: trippy_${{ matrix.target }}_${{ needs.create-release.outputs.trip_version }}_${{ env.DPKG_ARCH }}.deb
```

If you can update the PR based on this (and squash it down to a single commit) then we should be good to merge it.

**Comment by fujiapple852** [comment]
Merged, thank you for your contribution @Marwanmhks !

**Pull request merged**